chain/store/store.go

```diff
@@ -863,10 +863,19 @@ func (cs *ChainStore) Contains(ctx context.Context, ts *types.TipSet) (bool, err
 func (cs *ChainStore) GetBlock(ctx context.Context, c cid.Cid) (*types.BlockHeader, error) {
```

Comment:
@virajbhartiya I don't think this is the right way to fix this. `GetBlock` is used across APIs and we shouldn't change it's behaviour.

This issue is specifically for the `EthGetBlockByNumber` API to return a (nil, nil) to the user for null rounds i.e. epochs for which the chain did not produce any block. See https://filecoinproject.slack.com/archives/CP50PPW2X/p1726047559393929 for what null rounds are.

Reply:
The fix is very simply. See https://github.com/filecoin-project/lotus/blob/13fb3223fcc9212dd7349c26cc782d0a5dc377ca/node/impl/full/eth_utils.go#L91.

`getTipsetByBlockNumber` already returns an `ErrNullRound` (would be good learning to understand how/why it determines an epoch to be a null round). If `getblockByNumber` sees this error -> it should return no block and no error to the user.

Let's also do some digging in the itests to see how we can identify and query for null rounds to assert this behaviour.

Reply:
Oh okay, thanks for clarifying, I'll revert the changes made to `getBlock` and fix `ethGetBlockByNumber` to return accordingly if the epoch is a null round

Reply:
This is how I thought of implementing the test
```go
func TestEthGetBlockByNumberNullRound(t *testing.T) {
	ctx, cancel, client := kit.SetupFEVMTest(t)
	defer cancel()

	// Find a null round
	head, err := client.ChainHead(ctx)
	require.NoError(t, err)

	var nullRound int64
	for i := head.Height(); i > 0; i-- {
		ts, err := client.ChainGetTipSetByHeight(ctx, i, types.EmptyTSK)
		require.NoError(t, err)
		if len(ts.Blocks()) == 0 {
			nullRound = int64(i)
			break
		}
	}

	if nullRound == 0 {
		t.Skip("No null round found in recent history, skipping test")
	}

	// Get the block for the null round
	block, err := client.EthGetBlockByNumber(ctx, strconv.FormatUint(uint64(nullRound), 10), false)
	require.NoError(t, err)
	require.Equal(t, ethtypes.EthBlock{}, block, "Expected empty block for null round")
}
```
But while running it is unable to find a null round. Is there some way to force a null round to test it out?
@aarshkshah1992 @rvagg

Reply:
@virajbhartiya You can see how null rounds can be injected at https://github.com/filecoin-project/lotus/blob/13fb3223fcc9212dd7349c26cc782d0a5dc377ca/itests/eth_transactions_test.go#L431.

I just searched through the itests folder to find this.